flathub/website
# Backend payments

This adds support for building Stripe transactions and for vending apps.  It is incomplete.  At initial push of the PR it is missing UI for vending setup and making a purchase; and the final link from completion of purchase to ownership of an app token is not done.

To test and work on this, follow the stripe account setup instructions in `backend/README.md` and then you can do the following:

---

1. Login
2. Create stripe express account
3. test phone number, USA
4. Test code
5. Individual
6. Enter name, address, etc.  Use fake numbers
7. Individual
8. Software, click in url box, choose 'product description', enter some text
9. Use test account
10. Update additional information
11. Use test documents and save it all, this if everything is good, hit done.

An address to use is:

9601 Wilshire Blvd
/blank row/
Beverley Hills
California
90210

Next, visit the API page, find the `POST /vending/{appid}/setup` endpoint.
Run it for the test app and set up vending.

I used `org.flatpak.qtdemo` and:

```json
{
  "currency": "usd",
  "appshare": 70,
  "recommended_donation": 500,
  "minimum_payment": 0
}
```

This should return OK, with a description of the app split

next visit the `POST /vendingapp/{appid}` endpoint and run it for the test demo
passing in some amount of USD above five dollars

This will return a transaction ID, which you can see in your user's wallet page
and can complete a purchase from there.
